### DISCUSSION TITLE
Execute flow action on every item inside collection

### DISCUSSION THREAD

**Discussion by LasseRosenow** [directus/directus#16742] [Archive]
I am currently trying to implement a newsletter with directus flows.

I have a collection containing users that subscribed the newsletter.
And I have a collection containing the newsletter itself.

If I create a new newsletter, a flow gets triggered and is supposed to send an e-mail to every user inside the collection of users that subscribed it .

But I don't really find a way to build this iterating over the collection logic in the flows ui.

Is this possible? If not, is this planned?

**Comment by Guiqft** [comment]
Hi! Actually, you can achieve this using a little dirty trick suggested by a member on [this Discord thread](https://discord.com/channels/725371605378924594/1031621740822003795).
He posted a video about this: 
https://www.youtube.com/watch?v=BxrXQST-wzw

**Reply by LasseRosenow** [reply]
Thank you for your quick reply, this is already quite interesting, but it depends on having all the items as input value already.

My use-case is a bit different. My input value is only one newsletter post and then I somehow need to read all items in a newsletter-subscriber collection and send my post to them.

**Reply by br41nslug** [reply]
You can use the `read data` operation to read any data you need and pass it to a second flow or use a script operation to loop over the results.

**Reply by LasseRosenow** [reply]
What I need is some sort of `Loop` that has an input value, and then accesses every item in a configured collection and generates an output trigger for each item passing the input value and the item on.

Like this:
![Unbenanntes Diagramm drawio](https://user-images.githubusercontent.com/10547444/206238861-e86b66fc-dd0b-456d-bbd0-fe31781710bf.svg)

**Reply by LasseRosenow** [reply]
Oh it seems github didn't update fast enough, so  I did not see the answer of br41nslug yet :o

**Reply by br41nslug** [reply]
The `Trigger Another Flow` operation will run the second flow `forEach` item if it gets passed an array. https://docs.directus.io/cookbook/flows/flows-for-loop.html

**Reply by LasseRosenow** [reply]
Thank you very much, the `Another Flow` already seems really good, now I manage to trigger the flow for every subscriber.

But there is still one issue. I don't see how I can also pass my newsletter message along to the `Another Flow`, because I need to pass the array from the read data operation. If I want to combine them in the payload field, this would result in an object rather then an array
